### PR TITLE
[router] Tracing Middleware

### PR DESCRIPTION
**Pull request by IfSentient**
Added tracing middleware to the `router` package, which wraps every request in a new span and adds all HTTP attributes to the span.

This is a follow-up to the PR to add tracing, CC @joeblubaugh

### CODE REVIEW

File: plugin/router/middleware.go [hunk 1]
```diff
@@ -54,3 +57,30 @@ func NewCapturingMiddleware(f func(ctx context.Context, r *backend.CallResourceR
...
+		ctx, span := tracer.Start(ctx, "middleware")
+		defer span.End()
+		routeInfo := MatchedRouteFromContext(ctx)
+
+		resp := next(ctx)
+
+		query := ""
+		if s := strings.SplitN(req.URL, "?", 1); len(s) > 1 {
+			query = s[1]
+		}
```
Comment (radiohead) [inline plugin/router/middleware.go:76]: Why not use `req.URL.RawQuery`?

**Review by radiohead — APPROVED** [review]

**Review by ferruvich — APPROVED** [review]
LGTM 👍🏼 agreeing with Igor's comment about the usage of `RawQuery`

**Review by radiohead — APPROVED** [review]

File: plugin/router/middleware.go (hunk 1 above)
Comment (radiohead) [reply plugin/router/middleware.go]: Ah, of course! I completely forgot that we're not dealing with an `http.Request` here but rather a custom one.

Comment (joeblubaugh) [reply plugin/router/middleware.go]: was there any consideration of redacting information that's present in the query string?

**Pull request merged**